CMakeLists.txt

```diff
@@ -445,6 +457,13 @@ if(BUILD_ONNX_PYTHON)
                                            # unsigned from include\google\protob
                                            # uf\wire_format_lite.h
                                    ${EXTRA_FLAGS})
+    if(ONNX_USE_PROTOBUF_SHARED_LIBS)
+      target_compile_options(onnx_cpp2py_export 
+                             PRIVATE /wd4251 # 'identifier' : class 'type1' needs to 
+                                             # have dll-interface to be used by 
+                                             # clients of class 'type2'
+                            )
+    endif()
     target_compile_options(onnx_cpp2py_export PRIVATE /MT)
```

Comment:
No. It won't work.
See: line 467 in your change

Reply:
"/MT" causes the application to use the multithread, static version of the run-time library. Defines _MT and causes the compiler to place the library name LIBCMT.lib into the .obj file so that the linker will use LIBCMT.lib to resolve external symbols.

In this case, it must use "/MD"

Reply:
We did not want to change the default but the env var provides an option for /MD[d]

---
In reply to: [345602405](https://github.com/onnx/onnx/pull/2452#discussion_r345602405) [](ancestors = 345602405)